pyanaconda/network.py

```diff
@@ -318,13 +319,17 @@ def write_configuration(overwrite=False):
     task_proxy = NETWORK.get_proxy(task_path)
     sync_run_task(task_proxy)
 
-    task_path = network_proxy.InstallNetworkWithTask(overwrite)
-    task_proxy = NETWORK.get_proxy(task_path)
-    sync_run_task(task_proxy)
-
-    task_path = network_proxy.ConfigureHostnameWithTask(overwrite)
-    task_proxy = NETWORK.get_proxy(task_path)
-    sync_run_task(task_proxy)
+    if is_module_available(NETWORK) \
+            or network_proxy.Kickstarted:
+        task_path = network_proxy.InstallNetworkWithTask(overwrite)
+        task_proxy = NETWORK.get_proxy(task_path)
+        sync_run_task(task_proxy)
+
+    if is_module_available(NETWORK) \
+            or network_proxy.Hostname and network_proxy.Hostname != DEFAULT_HOSTNAME:
```

Comment:
I would move the conditions inside the DBus module or the DBus task if possible. Same above.